addons/mail/static/src/messaging/entity/entity/field.js

```diff
@@ -0,0 +1,796 @@
+         * Dependencies just need name of fields, but dependents cannot just
+         * rely on inverse field names because these dependents are a subset.
+         */
+        this.id = id;
+        /**
+         * This prop only makes sense in a relational field. This contains
+         * the name of the field name in the inverse relation. This may not
+         * be defined in declared field definitions, but processed relational
+         * field definitions always have inverses.
+         */
+        this.inverse = inverse;
+        /**
+         * This prop only makes sense in a relational field. If set, when this
+         * relation is removed, the related entity is automatically deleted.
+         */
+        this.isCausal = isCausal;
+        /**
+         * If set, this field acts as a related field, and this prop contains
+         * a string that references the related field. It should have the
+         * following format: '<relationName>.<relatedFieldName>', where
+         * <relationName> is a relational field name on this entity or a parent
+         * entity (note: could itself be computed or related), and
+         * <relatedFieldName> is the name of field on the entities that are
+         * related to current entity from this relation. When there are more
+         * than one entity in the relation, it maps all related fields per
+         * entity in relation.
+         */
+        this.related = related;
+        /**
+         * This prop only makes sense in a relational field. Determine which
+         * type of relation there is between current entity and other entities.
+         * 4 types of relation are supported: 'one2one', 'one2many', 'many2one'
+         * and 'many2many'.
+         */
+        this.relationType = relationType;
+        /**
+         * This prop only makes sense in a relational field. Determine which
+         * entity name this relation refers to.
+         */
+        this.to = to;
+    }
+
+    /**
+     * Combine current field definition with provided field definition and
+     * return the combined field definition. Useful to track list of hashes of
+     * a given field, which is necessary for the working of dependent fields
+     * (computed and related fields).
+     *
+     * @param {mail.messaging.entity.Field} field
+     * @returns {mail.messaging.entity.Field}
+     */
+    combine(field) {
+        return new Field(Object.assign({}, this, {
+            dependencies: this.dependencies.concat(field.dependencies),
+            hashes: this.hashes.concat(field.hashes),
+        }));
+    }
+
+    /**
+     * Perform computation of this field, which is either a computed or related
+     * field.
+     *
+     * @param {mail.messaging.entity.Entity} entity
+     */
+    doCompute(entity) {
+        if (this.compute) {
+            this.set(entity, entity[this.compute]());
+            return;
+        }
+        if (this.related) {
```

Comment:
Sorry, don't understand at all what you suggest here.

Only part of your comment I think I'm able to dissect is not having related value being stored on this entity, but here I think we want to have it stored so that observing this entity also observe this related relation.

I think an oral explanation is required, because this comment looks like a brainstorm on related fields, and I would be happy to discuss on that.